### PR TITLE
Add issue tracker support

### PR DESCRIPTION
**Pull request by timja**
Requires https://github.com/jenkins-infra/update-center2/pull/331

Ref: https://groups.google.com/d/msg/jenkinsci-dev/haFTYlhp7h8/hkApal9WAgAJ

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java [hunk 1]
```diff
@@ -27,6 +28,8 @@ public void parse(JSONObject pluginJson, Plugin plugin) {
     plugin.setSha1(pluginJson.optString("sha1", null));
     plugin.setTitle(pluginJson.optString("title", null));
     plugin.setUrl(pluginJson.optString("url", null));
+    JSONObject issueTracker = (JSONObject) pluginJson.opt("issueTracker");
+    plugin.setIssueTracker(new IssueTracker(issueTracker.getString("name"), issueTracker.getString("url")));
```
Comment (halkeye) [inline src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java]: You get issue tracker optionally, then use it immediately as if it's not null.

You need to confirm its not null before getting

Comment (halkeye) [reply src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java]: I think it's also causing the build to fail.

I'd also make sure there's a test that shows the data is loaded properly

Comment (timja) [reply src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java]: Update site will always set it after my change so it’s not really optional anymore, but may as well make it optional for safety I guess

Will take a look at the tests later

File: src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java [hunk 2]
```diff
@@ -27,7 +28,11 @@ public void parse(JSONObject pluginJson, Plugin plugin) {
     plugin.setSha1(pluginJson.optString("sha1", null));
     plugin.setTitle(pluginJson.optString("title", null));
     plugin.setUrl(pluginJson.optString("url", null));
-    plugin.setVersion(pluginJson.optString("version", null));
+    JSONObject issueTracker = (JSONObject) pluginJson.opt("issueTracker");
+    if (issueTracker != null) {
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java]: Personally I would use java optional 😅
```java
Optional<JSONObject> issueTracker = optional.ofNullable(pluginJson.opt("issueTracker"))
issueTracker.ifPresent(it -> {
    plugin.setIssueTracker(new IssueTracker(it.getString("name"), it.getString("url")));
    plugin.setVersion(pluginJson.optString("version", null));
})
```

File: src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java [hunk 3]
```diff
@@ -27,7 +28,11 @@ public void parse(JSONObject pluginJson, Plugin plugin) {
...
+      plugin.setIssueTracker(new IssueTracker(issueTracker.getString("name"), issueTracker.getString("url")));
+      plugin.setVersion(pluginJson.optString("version", null));
```
Comment (casz) [inline src/main/java/io/jenkins/plugins/generate/parsers/RootPluginDataParser.java]: why is this inside the if issueTracker?

**Comment by zbynek** [comment]
Plugin site gets the issue tracker directly from Update Center, so this is no longer needed.

**Pull request closed without merging**